### ISSUE TITLE
Add support for a mode where positionals are not positional

### ISSUE THREAD

**Issue by felixSchl** [felixSchl/docopt.js#24]
This is for convenience to the user:

```
usage: git branch [options] [-r | -a] [--merged | --no-merged]
   or: git branch [options] [-l] [-f] <branchname> [<start-point>]
   or: git branch [options] [-r] (-d | -D) <branchname>...
   or: git branch [options] (-m | -M) [<oldbranch>] <newbranch>
```

e.g. `git branch -m`, this all works:

`git branch -m foo bar`
`git branch foo -m bar`
`git branch foo bar -m`

This makes it easy to add another flag at then end should the user forget one.

**Comment by felixSchl** [comment]
Work is tracked on [`feature/free-positionals`](https://github.com/felixSchl/neodoc/tree/feature/free-positionals) and almost ready

**Comment by felixSchl** [comment]
Landed via 773941c7999a49cac2f813b50c38593a9ad5cc41
